Sprint-1/fix/median.js

```diff
@@ -6,9 +6,40 @@
 // or 'list' has mixed values (the function is expected to sort only numbers).
 
 function calculateMedian(list) {
-  const middleIndex = Math.floor(list.length / 2);
-  const median = list.splice(middleIndex, 1)[0];
-  return median;
+  if (!Array.isArray(list) || list.length ===0) {
+    return null
+  }
+  const onlyNumbers = list.filter(item => typeof item  === "number" &&  !isNaN(item))
+  if (onlyNumbers.length === 0) {
+    return null
+  }
+  const sortedList = [...onlyNumbers].sort((a,b) => a - b)
```

Comment:
Why do you do the `[...onlyNumbers]` here instead of just using `onlyNumbers`? What problem are you avoiding with it?